esp_lcd_panel_interface.h missing?

### The problem

I have an issue with compiling a project that uses FastLED. It seems that one of the libraries from the ESP toolchain is missing, a few months ago (I think around ESPHome 2025.10.0) it compilled without issues. I have tried both cleaning the project and cleaning all projects to no avail. The "fix" of changing to one of the other addressable components is not working because esp32_rmt_led_strip I think doesn't support this many LEDs (900) and the NeoPixelBus compiled but ended up crashing my ESP32S3 board.

The hardware is the ESP32-S3-ETH-8DI-8RO from waveshare with some additional components (DHT11 and a levelshifter for the LED output). The output I attached is from a compilation after a previous failed one, I can attach a  full one after a clean but the error itself is the same.

### Which version of ESPHome has the issue?

2026.2.4

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP32

### Component causing the issue

FastLED

### YAML Config

```yaml
esphome:
  name: blindswaveshare
  friendly_name: "MR Blinds"
  # min_version: 2025.5.0
  name_add_mac_suffix: false
  on_boot:
    then:
      - pcf85063.read_time

# Waveshare ESP32-S3-ETH-8DI-8RO uses ESP32-S3-WROOM-1U-N16R8
# -> 16 MB flash
# -> 8 MB psram
esp32:
  board: esp32-s3-devkitc-1
  flash_size: 16MB
  framework:
    type: arduino

psram:
  mode: octal
  speed: 80MHz

# Enable logging
logger:

# Enable Home Assistant API
api:
  encryption:
    key: "k87k0//8b0z/5tpAQapyUzMd1zp83DhYcUgJoGBlTyY="
  actions:
    # Ability to command the buzzer from Home Assistant
    - action: rtttl_play
      variables:
        song_str: string
      then:
        - rtttl.play:
            rtttl: !lambda 'return song_str;'

# Allow Over-The-Air updates
ota:
- platform: esphome

# wifi:
#   use_address: 172.16.2.6
#   networks:
#   - ssid: !secret wifi_ssid
#     password: !secret wifi_password
#     manual_ip:
#       static_ip: 172.16.2.6
#       gateway: 172.16.0.1
#       subnet: 255.255.0.0
#   - ssid: !secret extra_fallback_ssid
#     password: !secret extra_fallback_password
#     manual_ip:
#       static_ip: 172.16.2.6
#       gateway: 172.16.0.1
#       subnet: 255.255.0.0
#   ap:
#     ssid: "Blinds-and-Lights-Controller"
#     password: !secret fallback_ap_password

# text_sensor:
#   - platform: wifi_info
#     ip_address:
#         name: "${device_name} - IP Address"
#     ssid:
#         name: "${device_name} - Wi-Fi SSID"
#     bssid:
#         name: "${device_name} - Wi-Fi BSSID"
#   - platform: version
#     name: "${device_name} - ESPHome Version"
#     hide_timestamp: true

# Ethernet configuration - mutually exclusive with WiFi!
ethernet:
  type: W5500
  clk_pin: GPIO15
  mosi_pin: GPIO13
  miso_pin: GPIO14
  cs_pin: GPIO16
  interrupt_pin: GPIO12
  manual_ip:
    static_ip: 172.16.2.6
    gateway: 172.16.0.1
    subnet: 255.255.0.0
# text_sensor:
#   - platform: ethernet_info
#     ip_address:
#       name: "${device_name} - IP Address"
#       address_0:
#         name: "${device_name} - IP Address 0"
#       address_1:
#         name: "${device_name} - IP Address 1"
#       address_2:
#         name: "${device_name} - IP Address 2"
#       address_3:
#         name: "${device_name} - IP Address 3"
#       address_4:
#         name: "${device_name} - IP Address 4"
#     dns_address:
#       name: "${device_name} - DNS Address"
#     mac_address:
#       name: "${device_name} - MAC Address"

# Enable Web server
web_server:
  port: 80

# I2C bus
i2c:
  sda: GPIO42
  scl: GPIO41
  scan: false
  frequency: 100kHz
  id: i2cbus

pca9554:
  - id: 'TCA9554_hub'
    address: 0x20

# RS485 / Modbus
uart:
  - id: modbus_uart
    tx_pin: GPIO17
    rx_pin: GPIO18
    baud_rate: 38400
    stop_bits: 1 #default to 8E1
    data_bits: 8 #default to 8E1
    parity: EVEN #default to 8E1

# see: https://esphome.io/components/time.html
time:
  - platform: homeassistant
    id: homeassistant_time
    on_time_sync:
      then:
        # Update the RTC when the synchronization was successful
        pcf85063.write_time:
  - platform: pcf85063
    id: pcf85063_time

binary_sensor:
  - platform: status
    name: "Status"

  - platform: gpio
    name: "Boot Button"
    pin:
      number: 0
      ignore_strapping_warning: true
      mode:
        input: true
      inverted: true
    disabled_by_default: true
    on_press:
      then:
        - button.press: restart_button

# Digital Inputs
  - platform: gpio
    id: di1
    name: "DI1"
    pin:
      number: GPIO4
      mode: INPUT_PULLUP
      inverted: true
    filters:
      - delayed_on_off: 10ms

  - platform: gpio
    id: di2
    name: "DI2"
    pin:
      number: GPIO5
      mode: INPUT_PULLUP
      inverted: true
    filters:
      - delayed_on_off: 10ms

  - platform: gpio
    id: di3
    name: "DI3"
    pin:
      number: GPIO6
      mode: INPUT_PULLUP
      inverted: true
    filters:
      - delayed_on_off: 10ms
    on_state_change: 
      then:
        - light.toggle: rgb_lights
        - light.toggle: white_light

  - platform: gpio
    id: di4
    name: "DI4"
    pin:
      number: GPIO7
      mode: INPUT_PULLUP
      inverted: true
    filters:
      - delayed_on_off: 10ms

  - platform: gpio
    id: di5
    name: "DI5"
    pin:
      number: GPIO8
      mode: INPUT_PULLUP
      inverted: true
    filters:
      - delayed_on_off: 10ms

  - platform: gpio
    id: di6
    name: "DI6"
    pin:
      number: GPIO9
      mode: INPUT_PULLUP
      inverted: true
    filters:
      - delayed_on_off: 10ms

  - platform: gpio
    id: di7
    name: "DI7"
    pin:
      number: GPIO10
      mode: INPUT_PULLUP
      inverted: true
    filters:
      - delayed_on_off: 10ms

  - platform: gpio
    id: di8
    name: "DI8"
    pin:
      number: GPIO11
      mode: INPUT_PULLUP
      inverted: True
    filters:
      - delayed_on_off: 10ms

# buzzer
output:
  - platform: ledc
    pin:
      number: GPIO46
      ignore_strapping_warning: true
    id: buzzer


  - platform: gpio
    # name: "Relay 7"
    id: relay7
    pin:
      pca9554: TCA9554_hub
      number: 6
      mode:
        output: true
      inverted: false
    

  # - platform: gpio
  #   # name: "Relay 8"
  #   id: relay8
  #   pin:


rtttl:
  output: buzzer
  id: rtttl_buzzer
  gain: 30%

# RGB LED
light:
  - platform: fastled_clockless
    rgb_order: RGB
    chipset: WS2812
    # variant: ws2812
    pin: GPIO38
    num_leds: 1
    name: "RGB LED"
    id: rgb_led
  - platform: fastled_clockless
    
    rgb_order: RGB
    chipset: WS2812
    # variant: ws2812
    name: Color Lights
    id: rgb_lights
    # rmt_symbols: 192
    num_leds: 900
    pin: 21
    power_supply: rgb_psu
  - platform: binary
    name: "White Lights"
    id: white_light
    output: relay7
  
    
power_supply:
  - id: rgb_psu
    pin:  # Relay 8
      pca9554: TCA9554_hub
      number: 7
      mode:
        output: true
      inverted: false

# Relays
switch:
  - platform: gpio
    name: "Relay 1"
    id: relay1
    pin:
      pca9554: TCA9554_hub
      number: 0
      mode:
        output: true
      inverted: false
    

  - platform: gpio
    name: "Relay 2"
    id: relay2
    pin:
      pca9554: TCA9554_hub
      number: 1
      mode:
        output: true
      inverted: false
    

  - platform: gpio
    name: "Relay 3"
    id: relay3
    pin:
      pca9554: TCA9554_hub
      number: 2
      mode:
        output: true
      inverted: false
    
     

  - platform: gpio
    name: "Relay 4"
    id: relay4
    pin:
      pca9554: TCA9554_hub
      number: 3
      mode:
        output: true
      inverted: false
    

  - platform: gpio
    name: "Relay 5"
    id: relay5
    pin:
      pca9554: TCA9554_hub
      number: 4
      mode:
        output: true
      inverted: false
    

  - platform: gpio
    name: "Relay 6"
    id: relay6
    pin:
      pca9554: TCA9554_hub
      number: 5
      mode:
        output: true
      inverted: false

  - platform: template
    id: lights_but
    name: "Mike's room lights"
    internal: False
    turn_on_action: 
      then:
        - light.turn_on: rgb_lights
        - light.turn_on: white_light
  
    turn_off_action: 
      then:
        - light.turn_off: rgb_lights
        - light.turn_off: white_light

    

# Virtual buttons
button:
  - platform: restart
    name: "Restart"
    id: restart_button
    entity_category: config

  - platform: factory_reset
    name: "Factory Reset"
    id: reset
    entity_category: config

  - platform: safe_mode
    name: "Safe Mode"
    internal: false
    entity_category: config

    



cover:
  - platform: time_based
    id: blind_1
    name: "MR Blind Left"
    close_duration: 
      25 seconds
    open_duration: 
      25 seconds
    close_action: 
      then:
        - switch.turn_off: relay1
        - switch.turn_on: relay2
    stop_action: 
      then:
        - switch.turn_off: relay1
        - switch.turn_off: relay2
    open_action: 
      then:
        - switch.turn_off: relay2
        - switch.turn_on: relay1
  - platform: time_based
    id: blind_2
    name: "MR Blind Right"
    close_duration: 
      25 seconds
    open_duration: 
      25 seconds
    close_action: 
      then:
        - switch.turn_off: relay4
        - switch.turn_on: relay3
    stop_action: 
      then:
        - switch.turn_off: relay4
        - switch.turn_off: relay3
    open_action: 
      then:
        - switch.turn_off: relay3
        - switch.turn_on: relay4
  - platform: time_based
    id: blind_3
    name: "Living Room Blind"
    close_duration: 
      60 seconds
    open_duration: 
      60 seconds
    close_action: 
      then:
        - switch.turn_off: relay5
        - switch.turn_on: relay6
    stop_action: 
      then:
        - switch.turn_off: relay5
        - switch.turn_off: relay6
    open_action: 
      then:
        - switch.turn_off: relay6
        - switch.turn_on: relay5


sensor:
  - platform: dht
    pin: 1
    humidity: 
      name: "Attic Humidity"
    temperature: 
      name: "Attic Temperature"
    update_interval: 5s
```

### Anything in the logs that might be useful for us?

```txt
Processing blindswaveshare (board: esp32-s3-devkitc-1; framework: arduino, espidf; platform: https://github.com/pioarduino/platform-espressif32/releases/download/55.03.37/platform-espressif32.zip)
--------------------------------------------------------------------------------
HARDWARE: ESP32S3 240MHz, 320KB RAM, 16MB Flash
 - contrib-piohome @ 3.4.4 
 - framework-arduinoespressif32 @ 3.3.7 
 - framework-arduinoespressif32-libs @ 5.5.2 
 - framework-espidf @ 3.50502.0 (5.5.2) 
 - tool-cmake @ 4.0.3 
 - tool-esp-rom-elfs @ 2024.10.11 
 - tool-esptoolpy @ 5.1.2 
 - tool-ninja @ 1.13.1 
 - tool-scons @ 4.40801.0 (4.8.1) 
 - toolchain-xtensa-esp-elf @ 14.2.0+20251107
Reading CMake configuration...
Dependency Graph
|-- FastLED @ 3.9.16
|-- noise-c @ 0.1.10
Compiling .pioenvs/blindswaveshare/lib6b1/FastLED/platforms/esp/32/clockless_i2s_esp32s3.cpp.o
Compiling .pioenvs/blindswaveshare/lib6b1/FastLED/platforms/esp/32/i2s/i2s_esp32dev.cpp.o
Compiling .pioenvs/blindswaveshare/lib6b1/FastLED/platforms/esp/32/rmt_4/idf4_rmt.cpp.o
Compiling .pioenvs/blindswaveshare/lib6b1/FastLED/platforms/esp/32/rmt_4/idf4_rmt_impl.cpp.o
Compiling .pioenvs/blindswaveshare/lib6b1/FastLED/platforms/esp/32/rmt_5/idf5_rmt.cpp.o
Compiling .pioenvs/blindswaveshare/lib6b1/FastLED/platforms/esp/32/rmt_5/strip_rmt.cpp.o
Compiling .pioenvs/blindswaveshare/lib6b1/FastLED/platforms/esp/32/spi_ws2812/strip_spi.cpp.o
In file included from .piolibdeps/blindswaveshare/FastLED/src/third_party/yves/I2SClockLessLedDriveresp32s3/driver.h:9,
                 from .piolibdeps/blindswaveshare/FastLED/src/platforms/esp/32/clockless_i2s_esp32s3.cpp:19:
.piolibdeps/blindswaveshare/FastLED/src/third_party/yves/I2SClockLessLedDriveresp32s3/src/I2SClockLessLedDriveresp32s3.h:38:10: fatal error: esp_lcd_panel_interface.h: No such file or directory

***************************************************************************************
* Looking for esp_lcd_panel_interface.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:esp_lcd_panel_interface.h"
* Web  > https://registry.platformio.org/search?q=header:%1B%5Bm%1B%5BKesp_lcd_panel_interface.h
*
***************************************************************************************

   38 | #include "esp_lcd_panel_interface.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pioenvs/blindswaveshare/lib6b1/FastLED/platforms/esp/32/clockless_i2s_esp32s3.cpp.o] Error 1
========================== [FAILED] Took 9.73 seconds ==========================
```

### Additional information

_No response_

--- a/esphome/components/fastled_base/__init__.py
+++ b/esphome/components/fastled_base/__init__.py
@@ -7,6 +7,7 @@ from esphome.const import (
     CONF_OUTPUT_ID,
     CONF_RGB_ORDER,
 )
+from esphome.core import CORE
 
 CODEOWNERS = ["@OttoWinter"]
 fastled_base_ns = cg.esphome_ns.namespace("fastled_base")
@@ -41,5 +42,9 @@ async def new_fastled_light(config):
         cg.add(var.set_max_refresh_rate(config[CONF_MAX_REFRESH_RATE]))
 
     cg.add_library("fastled/FastLED", "3.9.16")
+    if CORE.is_esp32:
+        from esphome.components.esp32 import include_builtin_idf_component
+
+        include_builtin_idf_component("esp_lcd")
     await light.register_light(var, config)
     return var

--- a/tests/components/fastled_clockless/test.esp32-s3-ard.yaml
+++ b/tests/components/fastled_clockless/test.esp32-s3-ard.yaml
@@ -1,0 +1,1 @@
+<<: !include common.yaml

